### ISSUE TITLE
Jet Set Radio Future [PAL] [49470018]

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/game-compatibility#178]
## Issues
 - broken graphics on various locations

_From @furrya on March 21, 2017 0:58_

I just have launched the game on windows 7 x 64 , and I can see normal graphic on some locations , but  on others it's broken
![screenshot_7](https://cloud.githubusercontent.com/assets/25540784/24128013/d67b2838-0e24-11e7-92c9-6de9af49c550.png)
![screenshot_9](https://cloud.githubusercontent.com/assets/25540784/24128014/d67f0cbe-0e24-11e7-959f-211dfba0f757.png)
![screenshot_11](https://cloud.githubusercontent.com/assets/25540784/24128015/d681b87e-0e24-11e7-8e8f-0ed07449c0b9.png)

_Copied from original issue: Cxbx-Reloaded/Cxbx-Reloaded#251_

**Comment by PatrickvL** [comment]
_From @BenNottelling on March 21, 2017 5:18_

Looking like windows 7 is the best OS for this emulator

**Comment by PatrickvL** [comment]
_From @furrya on March 22, 2017 0:23_

With the latest commit, the graphics now is same  on Windows 7 and on Windows 10
![screenshot_4](https://cloud.githubusercontent.com/assets/25540784/24176730/25a78e62-0ee1-11e7-8695-2b8895a4bc56.png)
![screenshot_5](https://cloud.githubusercontent.com/assets/25540784/24176731/25a8708e-0ee1-11e7-986f-e2bce1962733.png)
![screenshot_7](https://cloud.githubusercontent.com/assets/25540784/24176732/25a96c1e-0ee1-11e7-9934-c5b480f2652a.png)

**Comment by PatrickvL** [comment]
_From @furrya on March 25, 2017 1:15_

With the latest commits - Win 10 x64 
![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/24318136/dfe13792-1143-11e7-8fb1-020f2c8f3d7d.png)

**Comment by PatrickvL** [comment]
_From @bhodg22 on March 25, 2017 11:56_

Does the rest of the level stay normal for you? I find it sometimes flickers to shades of black at times, also what GPU are you using

**Comment by PatrickvL** [comment]
_From @furrya on March 25, 2017 12:20_

@bhodg22 Same behaviour on my Palit Geforce 610, Regardless of the OS ( Win 7 x64, Win 10 x64) . Flickers appears many times on the start and ingame also

**Comment by PatrickvL** [comment]
_From @bhodg22 on March 30, 2017 9:4_

With the latest commit, I managed to get past the first cut scene and back into the garage, and the game was actually running full speed but then crashed when I tried going to the next level

**Comment by PatrickvL** [comment]
@bhodg22 : When reporting something like this, please mention exactly which commit you're referring to, by copy-pasting the URL for it, like this : `https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/2a8cc3722ba1eea8fe61c035149cd923f9ffe0a0` (which GitHub renders like this : https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/2a8cc3722ba1eea8fe61c035149cd923f9ffe0a0 )- this IS the one you're referring to, isn't it?

**Comment by PatrickvL** [comment]
@furrya This flickering is also present in other software, specifically the Dashboard, JSRF and other titles - see https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/issues/91#issuecomment-280917520

**Comment by PatrickvL** [comment]
_From @bhodg22 on March 30, 2017 9:52_

yeah thats the one i was referring to, I didnt bother posting any details because I tried to recreate it a second time but it didnt quite make it to the end of the cutscene, might try a third time soon

**Comment by PatrickvL** [comment]
_From @LukeUsher on March 30, 2017 9:56_

I've been able to play up to the point of the tutorial where you do the 'boost dash'. I usually get a crash at that point, but it seems to be caused by my graphics drivers.

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 18, 2017 17:43_

Urgent disaster! Needs remedying! After PR #375 (Overlay conversion) & #376 (Integrated copy OOVPAs. Most likely problem.) game crashes after company intros shown.

![2017-apr-19 _crash-after-intros](https://cloud.githubusercontent.com/assets/18428108/25144712/2feab868-24b2-11e7-8e94-3c1b659f0b41.png)
[KrnlDebug_(2017-04-19_Crash-after-intros).txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/930565/KrnlDebug_.2017-04-19_Crash-after-intros.txt)

**Comment by PatrickvL** [comment]
If the function locations differ, that's probably the main cause of regressing. Video issues shouldn't give such hard crashes

**Comment by PatrickvL** [comment]
_From @bhodg22 on April 18, 2017 22:53_

Strange, my JSRF runs fine on that same commit

**Comment by PatrickvL** [comment]
Then maybe you're running another version of the same game

**Comment by PatrickvL** [comment]
_From @bhodg22 on April 18, 2017 23:9_

Might be similar to the futurama issues, where other people can boot the PAL version as I still cannot

**Comment by PatrickvL** [comment]
_From @jarupxx on April 19, 2017 12:5_

@ObiKKa compare PR #374 log & #376, we will find the cause of the regression

**Comment by PatrickvL** [comment]
_From @LukeUsher on April 19, 2017 12:19_

JSRF also still works for me. How many times did you try? JSRF only boots successfully 8/10 times for me, the other times I get a similar error you see above.

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 19, 2017 15:46_

@LukeUsher & @jarupxx, that's strange. today I restarted the merged master of PR 374 (b525828e6959bd2f57b260e784b77391cb8c3024) coz I already downloaded it before. Still worked. Then I booted up the merged master of PR 376 (77933ad8ebf261634cfc2bde978f4fb8cc14d689), the old download - could be before you made that 2nd commit after I read the page (	OOVPA integration D3D 5849 to 3925).
But never mind that, in that build I ran JSRF (US region) again and it got to the menu and in-game again.

To Luke, I only ran JSRF twice on that last build (PR 376) a day before as I was testing other games.
Seems strange that some games do fail to run forward to the same progress line for inexplicable reasons sometimes. But, yeah, all seems to be fine. I'll test the newest build again, although that's unlikely to change my experience in Innocent Tears, etc.

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 20, 2017 23:2_

Oh, after about 4 crashes (That's a lot. Luke says his own test crashes on avg 2x every 10 tries.) I was able to see the main menu screen. But lots of blackness in the background and occasionally switching to yellow/white visuals.
Then in-game, first cutscene looks similar like before. But in-game was horrible. It appears that PR #384 (Implement HLE cache) may have mucked up the visuals for me. I think it may remember the first or so crash and then made the visuals worse on later boots. It's been much black over everything in the world for half of the viewing angles. Also when it's finally back to yellow and white polygons (corrupt vertex shaders ATM) I saw one gray shader on a platform was flickering.

![collage-2017-04-21](https://cloud.githubusercontent.com/assets/18428108/25256012/37bf8eea-2671-11e7-90bb-4fda6c6339dd.png)

* Re-test 22/04/2017: Thank you, Luke for that PR #389 (Fix HLE Cache) which included an option to clear HLE cache. That should become useful when testing various games to ensure that a previous bootup doesn't in any way affect sequential ones if one doesn't want that to happen.
Of course, this game has returned to 'normalcy' in buggy terms.

But after that PR #390 (DXBX improvements, etc?) the character model hasn't changed at all. Also none of us the testers are the wiser as to why this game slows down after starting to get into the game. Is it the memory leaks?

**Comment by PatrickvL** [comment]
I suspect the graphics are damaged because the HLE cache might be skipping the detection of render state and texture state addresses

**Comment by PatrickvL** [comment]
_From @LukeUsher on April 21, 2017 4:42_

Sorry about that, i overlooked that part of HLE patching. Should be an easy fix, I'll hopefully get that done before Monday.

**Comment by PatrickvL** [comment]
_From @furrya on April 22, 2017 2:39_


For me the state of the emulation gets better than in previous builds
The game looks better , it's looks like c1e4d973 now 

![screenshot_1](https://cloud.githubusercontent.com/assets/25540784/25300693/3869aab2-2750-11e7-960b-7cf753b87053.png)
![screenshot_2](https://cloud.githubusercontent.com/assets/25540784/25300696/3c825a9a-2750-11e7-968c-e1669ad9c7c8.png)
 1 st screen 18 Apr Build   and second Cxbx-Reloaded Version 1a9b20ad (Apr 21 2017)

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 23, 2017 23:1_

The main character model in the intro game cutscene seems to be fully rendered a bit better after PR #392 (Removed CreatePalette) and #393 (Temporarily disabled vertex buffer cache). And I could see some vehicles in the background in the main menu. But yeah loading the game takes a way long time and game runs slower as a result.
This game is easier said than done to fix up...

![collage-2017-04-24](https://cloud.githubusercontent.com/assets/18428108/25318169/7ee1f882-28cc-11e7-9695-0834ad36fdd1.png)

**Comment by PatrickvL** [comment]
_From @bhodg22 on April 24, 2017 0:36_

I also have the very slow loading and also the main menu is very slow, graphics on characters has improved alot and ingame is a bit slower, however my speedup trick still works to get the game running fullspeed

**Comment by PatrickvL** [comment]
This should now render a lot better!

**Comment by PatrickvL** [comment]
_From @ObiKKa on April 29, 2017 22:9_

No. After PR #406, it has crashed many times for me. I also re-cleared the HLE cache a few times later on as well.
Now the emulation seems to be so accurate that anything a game requires is missing then it becomes so brutal to said emulation!

![2017-apr-30 _index-buffer_crash_01](https://cloud.githubusercontent.com/assets/18428108/25559393/49c8fa42-2d7c-11e7-924d-8ea6b2133589.png)
[KrnlDebug_(2017-04-30_Index-Buffer_Crash).txt](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/files/966438/KrnlDebug_.2017-04-30_Index-Buffer_Crash.txt)

* Re-test: After using Luke's PR #410 (Revert of some AV query capabilities) I can now get in-game. Though the game would crash before entering the start menu a few times. So you have to clear the HLE cache before trying again, which will test your patience.
But I now see that my character's head in the intro cutscene is corrupted again.

**Comment by PatrickvL** [comment]
You're getting the same results as me, while Luke is getting increasingly better results. I suspect it's related to the version we're running (NTSC/PAL , older/newer builds of the game, or whatever).

**Comment by PatrickvL** [comment]
_From @blueshogun96 on April 29, 2017 22:21_

This is the exact same crash that I would get with the old Cxbx.  Something has definitely regressed.  It's the same with both PAL and NTSC, so it's not just him.

Several other titles (most notably SEGA titles) are regressing right now, and I'll make a note of each one.

Shogun.

**Comment by PatrickvL** [comment]
_From @LukeUsher on April 29, 2017 22:23_

Interestingly JSRF still works on my machine, in fact it's what I perform most of my testing with: I know how prone to regressions it is. Perhaps there's something different between my master and this one.

**Comment by PatrickvL** [comment]
_From @blueshogun96 on April 29, 2017 22:25_

Sadly, I'd have to assume that also.  It also reminds me of how Outrun2 appears to work for everyone else... except me!  I'll be sure to keep you updated.

Shogun.

**Comment by PatrickvL** [comment]
_From @LukeUsher on April 29, 2017 22:36_

Should be fixed now, my new implementation of AvQueryAvCapabilities was the problem.

**Comment by PatrickvL** [comment]
That was quick Luke - thanks!

**Comment by PatrickvL** [comment]
@blueshogun96 Have you considered submitting pull requests? Bug reports are nice and all, but better code, yeah, shiver me timbers, that's what we really want! ;)

**Comment by PatrickvL** [comment]
_From @blueshogun96 on April 29, 2017 22:47_

@PatrickvL I have, but unfortunately I haven't had my own fix for such things.  Some of these errors are beyond me, especially for me being out of the loop so long. :(

Shogun.

**Comment by PatrickvL** [comment]
_From @CakeLancelot on April 30, 2017 7:26_

In build 394aee6b the game is much more stable and I'm able to make it through the first cutscene, though the graphics are about the same compared to the last few builds. Seems a bit faster too but that may just be me. Excellent progress guys!
![cxbx_2017-04-30_02-20-22](https://cloud.githubusercontent.com/assets/5341136/25562492/6f388b42-2d4c-11e7-8a5e-6334912c2afa.png)
![cxbx_2017-04-30_02-23-05](https://cloud.githubusercontent.com/assets/5341136/25562491/6f376564-2d4c-11e7-9fa1-c47ff32cdc74.png)

**Comment by PatrickvL** [comment]
_From @LukeUsher on April 30, 2017 7:46_

In my recent testing, I managed to get past the cutscene and into gameplay properly, I played up until meeting Beat. One thing that was a little strange: after the intro finished, the music and sound became perfect.

**Comment by PatrickvL** [comment]
_From @CakeLancelot on April 30, 2017 8:38_

> In my recent testing, I managed to get past the cutscene and into gameplay properly, I played up until meeting Beat. One thing that was a little strange: after the intro finished, the music and sound became perfect.

I haven't played past dogenzaka hill, but during the time on that level the audio was still broken for me. 

On top of that as mentioned before that some of the particles and effects start working after using graffiti however this will revert when you enter the next 'scene' (includes the jumpcuts in cutscenes while loading in assets or whatever it's doing). 

Also encountered this weirdness when spraypainting a wall:
![cxbx_2017-04-30_02-37-32](https://cloud.githubusercontent.com/assets/5341136/25562839/aaceddd8-2d55-11e7-9734-56738b7796ad.png)
Then again as #271 shows there are differences in accuracy depending on graphics hardware, perhaps drivers even.

Let me know if you need any other info, my specifications are listed in #271.

**Comment by PatrickvL** [comment]
_From @ObiKKa on May 2, 2017 16:40_

Unfortunately I'm not able to continue in the Start menu for a while after all these little and big PRs. One same error message shows up before loading those intro splash screens (Press _Ignore_ to resume) and repeatedly after a brief period of time in the Start menu (I cannot see the menu buttons at this time).
I think it might be worth it to tackle that 'XInputPoll' problem.

Also congratulations to @blueshogun96 for his work on getting those Panzer Dragoon game(s) up and running! :) Hopefully a fourth regular programmer on this team? Dream on!

![2017-may-03 _error-box_01](https://cloud.githubusercontent.com/assets/18428108/25628514/d0e45c36-2fa9-11e7-9a1c-2e29e995ba8f.png)

* Re-test with Luke's recent commit:
Yes, JSRF did get back into the game after Luke's PR #424 (Remove unnecessary breakpoint. Reason: Resources are now separated from host resources so those errors kept popping up early.). But only after I tested 2 other games.
However I now see those stretched coloured graphical glitches linking the boots of my skater on a level surface as well as between the outer of either boot/leg to their adjacent arm when jumping high in the air.

Now, to the other two games - Namco Museum and Battlestar Galactica: No, those new error messages still pop up at the same time so they are not solved yet.

Created with photocollage.com:
![collage-2017-05-03](https://cloud.githubusercontent.com/assets/18428108/25638549/2186da48-2fcc-11e7-84ab-ccfd1a76c635.png)

* Re-test 2 (04/05/2017):
Hi again. Some slightly good news. The polygonal surface 'linking' errors between the appendages/limbs of the player model have been for the most part removed. But I still see that gray/white thin stretch between a couple limbs on the left side. I had to jump high in the air to capture a shot and show it clearly to you.
Now, I don't know which commit improved this glitch somewhat.
![2017-may-04_test3 _01_gfx-glitch](https://cloud.githubusercontent.com/assets/18428108/25675690/c27fc3f4-3082-11e7-9d5f-424c66656cf8.png)

* Re-test 3 (05/05/2017):
Even after a series of PR's consisting of logging improvements (ie. #436) Namco Museum and Battlestar Galactica still crashes the same as before. JSRF does have problems trying to reach the menu before I could get in-game (coz of similar error messages as other games). Also I saw the linking glitch between the boots like in re-test 1 above - it's perhaps a random area each time you boot the game up. Also the blue flashing stars upon grinding on the rails seem to be rendered worse.

**Comment by PatrickvL** [comment]
I've seen almost the same error for a while, until I fixed a bug in D3DSurface_LockRect; it didn't account for textures, instead of surfaces being passed in

**Comment by PatrickvL** [comment]
Regarding the glitches linking some elements together: this might be caused by the same issue that seems to have broken Turok (right after starting a game), probably an index buffer change. @LukeUsher Could it be that our index buffers are not correctly aligned?

**Comment by PatrickvL** [comment]
_From @bhodg22 on May 13, 2017 3:4_

Was meant to post this earlier, I managed to get to the beat mission in a build from about 2 months ago and the sound was also became perfect back then as well. As for one of the latest builds (Maybe from about 10 days ago?) Alot of the ingame elements have become invisible for me (buildings, the ground, rails, etc) making it impossible for me to progress further in the games as i cant see the levels!

**Comment by PatrickvL** [comment]
_From @CakeLancelot on May 13, 2017 7:3_

As of 2d720a1 the shading on the characters has improved, though many previous graphical issues are still present. 
![cxbx_2017-05-13_01-59-10](https://cloud.githubusercontent.com/assets/5341136/26023394/1f7c558c-3780-11e7-923e-a8b5fcaadbef.png)
vs 394aee6 
![cxbx_2017-04-30_02-20-22](https://cloud.githubusercontent.com/assets/5341136/26023397/2ae5caa2-3780-11e7-9a0e-d3049b89dd23.png)

**Comment by PatrickvL** [comment]
Which one is the better one? Do you have some reference material perhaps?

**Comment by AustinHnn** [comment]
<img width="960" alt="chrome_2017-08-23_02-02-40" src="https://user-images.githubusercontent.com/11362178/29575215-2d80b9c6-87a7-11e7-99e6-22330aa131d4.png">
Not the exact scene but the first one seems the most similar.

**Comment by HyperScorpio** [comment]
I can confirm a nother crash everytime we enter a new map location and visit the garage for 8 times. The loading stops and appears black screen with the error message to press ok or debug. The reason i'm saving the game many times to bypass the story.

**Comment by AustinHnn** [comment]
https://www.youtube.com/watch?v=6K3ECkhgO64 Take a look at this! Very interesting indeed,

**Comment by HyperScorpio** [comment]
@AustinHnn That thing happens to me as well. And the funny thing you have a Ryzen CPU and appears Black in the map, and i have a Intel CPU and all appears white. 😆 And that part when the graphics are fixed i can see the map perfectly, when you go back it shows white. 😆

**Comment by CakeLancelot** [comment]
@HyperScorpio @AustinHnn Please try to keep conversation to the gitter.im channel or the discord server, these issues have more or less already been reported. I will create a new issue to centralize everything related to JSRF soon, just need to make comparison images/videos.

**Comment by HyperScorpio** [comment]
@CakeLancelot Send us the links.

**Comment by PatrickvL** [comment]
@HyperScorpio Read the readme. It exists exactly for these type of questions.

**Comment by PatrickvL** [comment]
Right on top, you'll find a "chat on gitter" button. Pressing that should direct you to the chat room.

**Comment by PatrickvL** [comment]
Also, we do like to be treated nicely. No faul language, and using friendly expressions like "please" and "thank you" are appreciated

**Comment by HyperScorpio** [comment]
I finish the game and i save before crash, and i got the error on the loadings.

![1](https://user-images.githubusercontent.com/30736913/29684682-c9b8c482-890a-11e7-80bb-46c880483b4e.png)

Here's XBE file: [Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1249999/Xbe.txt)

**Comment by AustinHnn** [comment]
@CakeLancelot Not having a conversation. Was highlighting a graphical issue which this thread is for.

**Comment by CakeLancelot** [comment]
## Copying some information from #174:


When I was testing out JSRF I saw some differences in my game's rendering compared to the one posted. Now I still get the graphical errors as described in Cxbx-Reloaded/Cxbx-Reloaded#251 but my UI seems to be completely black at all times. 

At first I thought this difference was due to my graphics card but I've also tried running it on my iGPU and got the same results. So this might be something else, maybe OS.

From Cxbx-Reloaded/Cxbx-Reloaded#251 (UI renders correctly)
!["Correct" Rendering](https://cloud.githubusercontent.com/assets/25540784/24176732/25a96c1e-0ee1-11e7-9934-c5b480f2652a.png)

Running on my PC (UI is blacked out)
![cxbx_2017-03-24_18-56-34](https://cloud.githubusercontent.com/assets/5341136/24317278/a68a4c8e-10c3-11e7-831c-94857d816535.png)


Here are my PC's specs:

> MSI Z97-G55 SLI
Intel Pentium G3258 OC'ed to 4.4 ghz
Sapphire Nitro R9 FURY 4GB (Stock)
2 Crucial Ballistix Sport 8GB (total 16GB)

I'm running on Windows 8.1
Build db7da95d (Will update if needed)

**Comment by BenNottelling** [comment]
I'm getting the same back UI on my RX 480, haven't bothered to test iGPU

**Comment by CakeLancelot** [comment]
## Copying some information from Cxbx-Reloaded/Cxbx-Reloaded#251
> Which one is the better one? Do you have some reference material perhaps?

Sure, here's some comparisons:
![comparison](https://cloud.githubusercontent.com/assets/5341136/26023749/d8eb8148-3788-11e7-8323-5b1840147d8a.png)
![comparison2](https://cloud.githubusercontent.com/assets/5341136/26023752/ddf6571c-3788-11e7-8154-a63d873d2559.png)


I also noted that the lighting itself is more accurate, for example the first comparison.

**Comment by CakeLancelot** [comment]
There is also some information regarding graphical glitches in #44, though I have closed it per the one-game one-issue rule. Here are some highlights:

_From @CakeLancelot on August 8, 2017 18:8_

> Can you verify on hardware if possible?

I can confirm this issue (Second image is actual xbox):
![cxbx_2017-08-08_13-07-07](https://user-images.githubusercontent.com/5341136/29087166-a80c9e8a-7c3a-11e7-86ca-112b1dee72c8.png)
![graffiti](https://user-images.githubusercontent.com/5341136/29087180-adc70f86-7c3a-11e7-966d-2e9c78864814.png)




_From @HyperScorpio on August 10, 2017 14:14_

Map Zone - 99th Street.

![3](https://user-images.githubusercontent.com/30736913/29174484-61736bca-7dde-11e7-8469-fb01e892a069.png)

Graffiti problems.
![2](https://user-images.githubusercontent.com/30736913/29174487-65b60314-7dde-11e7-9949-df0ee7488ce2.png)

The annoying white cross.
![1](https://user-images.githubusercontent.com/30736913/29174491-67f5a18e-7dde-11e7-8e46-346cf615842a.png)

The annoying brown rectangle.
![4](https://user-images.githubusercontent.com/30736913/29174495-6ae009f2-7dde-11e7-9dea-699c8afd5dad.png)

After beat the crazy cop.
![5](https://user-images.githubusercontent.com/30736913/29174498-6cd30368-7dde-11e7-9641-7e07a1997fe2.png)

Before entering on the Garage.
![6](https://user-images.githubusercontent.com/30736913/29174501-6f04f614-7dde-11e7-929a-13cce6ca2f8c.png)

**Comment by CakeLancelot** [comment]
As of version a4fa5551 there appears to be a regression. I haven't tested each individual commit between this and 8b11cce5 as there aren't on appveyor but I will see if I can compile and test. The game crashes right after the final splash screen (graffiti warning). I have managed to get past this screen once but it then crashed promptly afterwards, not sure what the difference was there. Here is the error message and log:
```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x5E8875CF(=XInputPoll+0x5e6c2c52)

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Cancel   
---------------------------
```
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1316273/KrnlDebug.txt)

**Comment by LukeUsher** [comment]
Could you retest in #738?

IT might be necessary to clear your cache too (All EmuDisk partitions except for 1 and 2) in the Cxbx-Reloaded folder

**Comment by LukeUsher** [comment]
So the cause of JSRF instability in early boot appears to be that two function signatures are exactly the same,making it impossible to patch them accurately.

So my priority is going to be removing those patches.
```
8B4424048B0D........898130240000C2040090909090909090909090909090 00 0000 0020 _D3DDevice_SetVerticalBlankCallback@4 ^0006D ?g_pDevice@D3D@@3PAVCDevice@1@A
8B4424048B0D........89812C240000C2040090909090909090909090909090 00 0000 0020 _D3DDevice_SetSwapCallback@4 ^0006D ?g_pDevice@D3D@@3PAVCDevice@1@A```

**Comment by PatrickvL** [comment]
Those NOPs at the end don't help much either

**Comment by jarupxx** [comment]
They may look the same as 4361, but JSRF's 4134 has different signatures. More than that it is preferable that the patch becomes unnecessary.
```
8B4424048B0D........89812C240000C2040090909090909090909090909090 00 0000 0020 _D3DDevice_SetVerticalBlankCallback@4 ^0006D ?g_pDevice@D3D@@3PAVCDevice@1@A
8B4424048B0D........898128240000C2040090909090909090909090909090 00 0000 0020 _D3DDevice_SetSwapCallback@4 ^0006D ?g_pDevice@D3D@@3PAVCDevice@1@A```

**Comment by HyperScorpio** [comment]
I can´t bypass this error anymore, even rescan the dam game for x20 times. Thank you all 👍 
![error](https://user-images.githubusercontent.com/30736913/31181769-0f7091f6-a91a-11e7-9ba7-135dae59631b.png)

**Comment by LukeUsher** [comment]
It still works for me, I was playing it earlier today, perhaps you have a different version of the game? Or maybe the data is corrupted, or maybe you need to manually delete the Xbox Cache Partitions (All partitions except for Partition 1 & Partition2). There are a large number of possible causes here

**Comment by LukeUsher** [comment]
Also from your debug log it looks like you have an NTSC version of JSRF: This issue is for the version marked as region-other. 

You should create a new issue for that version of the game, if one doesn't exist already.

Also this thread should really be split up, it's impossible to tell which reports are for what version, it all got dumped in here!

**Comment by CakeLancelot** [comment]
> Also this thread should really be split up, it's impossible to tell which reports are for what version, it all got dumped in here!

I'll try to tackle this issue, I do agree it's very disorganized. [This should be a good reference for the different versions of the game](https://www.speedrun.com/jsrf/guide/wofi9). At the moment we have japan trial version and this issue, which I'll just specify as PAL. I also have the NTSC combo disc - the standalone is a lot less common, so I'll add that when possible.

**Comment by HyperScorpio** [comment]
@LukeUsher  thanks, i will check that

**Comment by HyperScorpio** [comment]
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1366977/KrnlDebug.txt)
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1366978/Xbe.txt)
[CxbxDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1366979/CxbxDebug.txt)

**Comment by Solstar** [comment]
Its way too fast!its unplayable for me!i have na i7 with 16gb ram and a geforce 1050,i tried the latest (14/11/2017) version and its stupidly fast that i cannot even see what's going on. what can i do to slow it down

**Comment by LukeUsher** [comment]
You can try turning on VSync in the config

**Comment by PatrickvL** [comment]
Timing being (severely) off is a known issue right now

**Comment by Solstar** [comment]
@LukeUsher that seems to work..in the intro at least. because after the logos and disclaimer ends,the game abruply ends with "received exception code " etc with the main theme still playing in the background

**Comment by Solstar** [comment]
after clearing the cache it seems to work now. 60 fps steady, no slowdown whatsoever. If it wasn't for those graphic bugs and ligthing issues, it would be perfectly playable.

**Comment by MewMew7** [comment]
For me the emulation is about is consistent as pigeon crap, and I have no idea why. I tried cleaning up the HLE cash, but it doesn't seem to have any sort of impact. The game just randomly starts up after a few attempts, without any apparent reason. Otherwise the game runs extremely smoothly (albeit with all the known graphical bugs).
![glitch](https://user-images.githubusercontent.com/34717730/34220948-69e5434a-e5c7-11e7-82fb-37bf30b969e8.png)
![noglitch](https://user-images.githubusercontent.com/34717730/34221452-209df3c4-e5c9-11e7-9e6f-54a55c20d233.png)

[KrnlDebug_glitch.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1576550/KrnlDebug_glitch.txt)
[KrnlDebug_noglitch.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1576579/KrnlDebug_noglitch.txt)
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1576580/Xbe.txt)

**Comment by Nathalie9** [comment]
i'm on win.10 /64bits i7 gtx 1060 Can not pass the loading screen every time I have this message on the black screen I also want to extract files with win.rar Magique.ISO and C-xbox 2.0.7. always the same result

![cxbx](https://user-images.githubusercontent.com/34754284/34275975-c3ce22a4-e69f-11e7-8907-e4c9a3a9b773.png)

[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1580742/KrnlDebug.txt)

[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1580743/Xbe.txt)

**Comment by CakeLancelot** [comment]
> I also want to extract files with win.rar Magique.ISO and C-xbox 2.0.7. always the same result

Don't use Winrar for extracting Xisos. Qwix or C-Xbox Tool should work, and if they don't the dump is probably damaged in some other way. I've also seen this error when VSync is not enabled, so maybe try doing that and see if it works.

**Comment by Nathalie9** [comment]
i have test VSync forced ON and i have same result :(

**Comment by LukeUsher** [comment]
Rather than extracting from an ISO, you could try using an FTP client to transfer the files directly from your Xbox console, direct from the D: drive if using.a disc. This will rule out issue switch the extraction tools you are using. Failing that, your game dump could be bad: can you confirm it works on a real Xbox too? 


Also you could try cleaning your cache partitions: this is different from clearing the HLECache. There is an EmuDisk sub folder under %AppData%\Cxbx-Reloaded. Please delete all contents of EmuDisk, except for Partition1 and Partition2: these ones should stay, and try again.

**Comment by Nathalie9** [comment]
My disk is not in very good condition its must be the main reason a little too much damage and my xbox original is dead :(

**Comment by ObiKKa** [comment]
New video from InsertMoreCoinsblog IMC: [Cxbx WIP (build from november) - Jet Set Radio Future (USA) - Shortplay](https://www.youtube.com/watch?v=iZZdx9iGJpE) (Dec 6, 2017).

**Comment by Photon-Phoenix** [comment]
Upon entering the first cutscene when entering Dogenzaka Hill, cxbx-reloaded issues to release a kernal dump to github and K's facial mesh stretches across the screen as well as an invisible body with only moving arms and hands visible.  Happens with or without pixel shaders.


![mesh](https://user-images.githubusercontent.com/2118535/34894217-7c218dd6-f7ae-11e7-8cad-866ba5ffa48b.png)


[JSRFPALStandalone.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1627416/JSRFPALStandalone.txt)
[KrnlDebugJSRFPALStandalone.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1627418/KrnlDebugJSRFPALStandalone.txt)

**Comment by Suplex7** [comment]
How do I slow my game down? It's speeding up all over the place.

**Comment by PatrickvL** [comment]
Timing is still being worked on

**Comment by HyperScorpio** [comment]
I found something interresting... If you go to the Garage and select the 3rd training and if you don´t complete the training, try to go around the Garage Map, it doesn´t show any bug or even in the sky, just thinking my self there is some problems with the charathers polygons and the graffiti Souls. You need to investigate more about that PatrickvL. 👍 Or there is some map far away from the garage just making this annoying bug. Cheers.

**Comment by zeraoo98** [comment]
I can confirm the last comment, tho I just had to first complete the 1st training and on my second (retry) of the same training, all the white bugs disappeared UNTIL I used the spray button, which launched the spray/graffiti animation with the "spray particles" effect. Right at that moment, all the massive white texture bugs came right back. Some ideas on this?

**Comment by ObiKKa** [comment]
Ingame still but this new video from JGG seems to suggest that the 'white-out' problem with the in-game GFX has been stopped but that the framerate has grinded almost to a halt.

- [Cxbx-Reloaded Xbox Emulator - Jet Set Radio Future Ingame! (fb5436e4)](https://www.youtube.com/watch?v=OqZLu0mHcXg&t=0s&list=PL4PD8cV5ypNDPTJbrol7A-5rvuteHWWYj&index=1) (John GodGames Emus. Published on 1 Mar 2018.)

**Comment by HyperScorpio** [comment]
Now works fine and good, but if we paint the start save, the white thing appears constantly not so worry about it. 
It Doesn´t mean you can´t fix it at all. 👍 
I'm using windows 10.

![sem titulo](https://user-images.githubusercontent.com/30736913/36926325-14df0a62-1e6f-11e8-9341-ccf126452de2.png)

**Comment by HyperScorpio** [comment]
A nother crash found it. I can´t bypass this.
Thank you all! 

![sem titulo](https://user-images.githubusercontent.com/30736913/36950710-08ead748-1ff2-11e8-85a1-011072d88e5c.png)

**Comment by CakeLancelot** [comment]
> A nother crash found it. I can´t bypass this.
> The older versions before 1 March was working good.
> Thank you all!

When does the crash occur exactly?

**Comment by HyperScorpio** [comment]
@CakeLancelot After we defeat Hayashi the in the stadium.
When he speaks "What i'm thinking...?" the error appears.
https://www.youtube.com/watch?v=udBHidVrDSw

**Comment by LukeUsher** [comment]
Can you please zip and upload your savedata so we can easily reproduce & investigate: 

Go to %AppData%/Cxbx-Reloaded/Partition1/UDATA/<TITLE_ID>

Where Title ID is  the title Id for your version of JSRF, you can find it in the Xbe dump.

**Comment by HyperScorpio** [comment]
@LukeUsher 
Here: [UDATA.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/1779172/UDATA.zip)

**Comment by HyperScorpio** [comment]
The newest releases the error continues, good luck with the fix.

**Comment by Rdup222** [comment]
After trying to play, All I can say is that it is currently in a playable state. 

But mind you, Theres a few sounds that are not working, like voicelines, Spray Sounds. ETC.

And from what the user Hyper Scorpio has said, The game stops after we beat hayashi again.

**Comment by HyperScorpio** [comment]
I need to play a new game, now DJ Professor K Shows even better. :)

**Comment by HyperScorpio** [comment]
Tested again and still crashes, i got something before i defeat Hayashi. DJ Professor K when he speaks the last ballon message just changed to french language. I will send 2 saves. The first one is all levels untill there with all Graffiti Souls completed and collected. The 2nd save is before i fight against Hayashi.
Please if you guys can fix it to able to continue to play.
Thank you all! 

![sem titulo](https://user-images.githubusercontent.com/30736913/37186307-b766d37e-233c-11e8-8648-170030e230e7.png)

[Jet Set Radio Future-8f0bb158.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/1795389/Jet.Set.Radio.Future-8f0bb158.zip)

[UDATA.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/1795385/UDATA.zip)

**Comment by HyperScorpio** [comment]
I just forget to post the french bug letters.

![french bug](https://user-images.githubusercontent.com/30736913/37215344-9719c9dc-23af-11e8-9bd7-04626fc3bb3a.png)

**Comment by HauruI** [comment]
Hi here, it's great to see the work accomplished ! Now my game runs perfectly fine talking to FPS...
The last bug I've seen comes when I paint graffitis... They don't appear:
![image](https://user-images.githubusercontent.com/37268710/37254742-6a8e3a76-2542-11e8-872d-46b24e359830.png)
Tried almost everything.

My specs:
Intel i5-7300HQ @2.5-3.5GHz
NVidia GTX 1050 @4GB GDDR5
8GB DDR4-2400 RAM
Jet Set Radio Future PAL 
Cxbx-Reloaded build b8c5808a

**Comment by HyperScorpio** [comment]
@HauruI they removed for a reason, need to be fixed.

**Comment by SputnikKaputtnik** [comment]
JSRF PAL used to be able to go ingame for me, but it regressed to instantly crashing a couple of months ago. I still keep the build from Nov 27, 2017 around because it's able to start both JSRF and Panzer Dragoon Orta.

I'm running an I7, GTX 1080 and Win10 64 bit

Here are the log dumps:

[KrnlDebug JSRF.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1803227/KrnlDebug.JSRF.txt)

[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1803233/Xbe.txt)

**Comment by SputnikKaputtnik** [comment]
@HyperScorpio What do you mean no difference? I ran both versions today. The older one will start the game, the new one will crash instantly.

**Comment by HyperScorpio** [comment]
@SputnikKaputtnik Strange. i always remove the %AppData% CXBX Reloaded and the Folder on the regedit.

**Comment by SputnikKaputtnik** [comment]
@HyperScorpio Deleting the roaming/appdata/cxbx folder did the trick. Thank you for pushing me in the right direction!

**Comment by sgtsaughter** [comment]
Uw

**Comment by HyperScorpio** [comment]
I need is someone can able to start a new game and quickly try it again, i'm busy right now to do this again.

**Comment by HyperScorpio** [comment]
I Tested the game again, i clean all files including the configuration, i just start and play and i got the crash again. Total 2h 10 min. We can use 2 free tools, one is Registry Scanner to delete all files belongs as CXBX and File Searchy to remove all the CXBX Database.

**Comment by CakeLancelot** [comment]
Regarding the Hayashi bug, the first bad commit appears to be 179c158. As the commit was mainly pertaining to HLEDB_V2, I've uploaded two HLEcaches in a zip file. BAD is the one generated by build 179c158 and doesn't work, while GOOD is the build right before it (e252cdf) that does work.

[HLECache.zip](https://github.com/Cxbx-Reloaded/game-compatibility/files/1822686/HLECache.zip)

**Comment by CakeLancelot** [comment]
By tweaking my HLEcache ini file on the latest build I've confirmed that this is related to it, but I haven't found out which specific function yet.
![image](https://user-images.githubusercontent.com/5341136/37566435-28e52d24-2a87-11e8-922e-436f2ec08c59.png)
(Noise Tanks after beating Hayashi boss battle)

**Comment by PatrickvL** [comment]
Summarizing the differences between the BAD and GOOD HLECache files attached above by @CakeLancelot :
```
D3DCubeTexture_LockRect=193d00 // Was absent in BAD. In GOOD, this equals Lock2DSurface in BAD *and* GOOD?!?
D3DDevice_CreateImageSurface=18f2c0 // Was 18d710 in BAD (D3D_CreateStandAloneSurface)
// Absent in GOOD : D3DDevice_GetTexture2=18e990
D3DVolumeTexture_LockBox=193da0 // Was 18f120 in BAD. In GOOD, this equals Lock3DSurface in BAD *and* GOOD?!?
// Absent in GOOD : D3D_CreateStandAloneSurface=18f2c0
```

**Comment by HyperScorpio** [comment]
Thank you guys, i'm going to use this one untill you guys in the future fix everything.

**Comment by HyperScorpio** [comment]
Adding this 2 .ini files the crash continues.

**Comment by LukeUsher** [comment]
The hard crash after Hayashi has been solved as of this PR https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/pull/1000

Turns out that the cause was simply a patch that we neglected to remove after fixing D3DDevice_SetTexture to correctly update the internal state.

**Comment by bhodg22** [comment]
To further what @ObiKKa was mentioned earlier, I was able to get 60FPS easily from build in early March however after some recent commits, the game basically chuggs down to a 1-5 fps. When the game is not in focus the FPS returns to normal, but no sound plays.

I have uploaded another video which demonstrates this (For some reason audio still does not record). https://youtu.be/093Lyz8kRFc

I can grab logs if required

**Comment by Rdup222** [comment]
The only problem I have is the sounds having that loud pop every time.  Like selecting something, grabbing a graffiti soul.    And spraying paint.  I still don’t know if you could fix that.

**Comment by HyperScorpio** [comment]
@Rdup222 Keep in touch if someday the sound get's fixed 100%, for now let's enjoy the game story. 💯

**Comment by Rdup222** [comment]
![image](https://user-images.githubusercontent.com/37114776/37620300-31ee5358-2b92-11e8-9de3-68212cede047.png)


Another problem is frame dipping on the intro, Missing textures on my side.

**Comment by CakeLancelot** [comment]
This issue got out of hand (more than 100 comments!) so I've created a more up to date issue [here](https://github.com/Cxbx-Reloaded/game-compatibility/issues/703). Please only post issues about the game there with detailed explanations. All other discussion can be in the [Discord](https://discord.gg/26Xjx23) or [Gitter ](https://gitter.im/Cxbx-Reloaded/Lobby)channel. Thank you.
